### PR TITLE
plat-mediatek: add support to extend MAX_XLAT_TABLE

### PR DESCRIPTION
**Pull request by ZheTing815**
We will create the memory pool from MEM_AREA_RES_VASPACE.
When using the reserved virtual memory space, it is necessary to increase MAX_XLAT_TABLE based on its size.

### CODE REVIEW

File: core/arch/arm/plat-mediatek/conf.mk [hunk 1]
```diff
@@ -14,6 +14,8 @@ CFG_DRAM_BASE ?= 0x40000000
 # default DRAM size 1 GiB
 CFG_DRAM_SIZE ?= 0x40000000
 
+CFG_MTK_RESERVED_VA ?= n
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-mediatek/conf.mk:17]: Please add a comment describing this new option.

**Comment by ZheTing815** [comment]
add some descriptions for the config

**Review by jenswi-linaro — COMMENTED** [review]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by ZheTing815** [comment]
tag applied

**Pull request merged**